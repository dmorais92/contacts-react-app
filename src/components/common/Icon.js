import React, { PureComponent } from "react";
import PropTypes from "prop-types";

import "./Icon.scss";

class Icon extends PureComponent {
  static propTypes = {
    icon: PropTypes.oneOf([
      "mobile",
      "geo",
      "user",
      "logo",
      "open",
      "close",
      "search"
    ]),
    size: PropTypes.oneOf(["s", "m", "l", "xl", "xs"]),
    color: PropTypes.oneOf(["primary", "secondary"])
  };

  static defaultProps = {
    icon: null
  };

  render() {
    const { icon, size = "m", color = "primary", ...otherProps } = this.props;
    switch (icon) {
      case "user":
        return (
          <i className={`Icon far fa-user ${size} ${color}`} {...otherProps} />
        );
      case "search":
        return (
          <i
            className={`Icon fas fa-search ${size} ${color}`}
            {...otherProps}
          />
        );
      case "logo":
        return (
          <i
            className={`Icon fas fa-user-circle ${size} ${color}`}
            {...otherProps}
          />
        );
      case "geo":
        return (
          <i
            className={`Icon fas fa-location-arrow ${size} ${color}`}
            {...otherProps}
          />
        );
      case "open":
        return (
          <i
            className={`Icon fas fa-angle-down ${size} ${color}`}
            {...otherProps}
          />
        );
      case "close":
        return (
          <i
            className={`Icon fas fa-angle-up ${size} ${color}`}
            {...otherProps}
          />
        );
      case "mobile":
        return (
          <i className={`Icon fas fa-phone ${size} ${color}`} {...otherProps} />
        );
      default:
        return (
          <i className={`Icon ${icon} ${size} ${color}`} {...otherProps} />
        );
    }
  }
}

export default Icon;
