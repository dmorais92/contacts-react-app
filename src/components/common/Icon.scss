@import '../../variables.scss';

.Icon {
  margin: 5px;
  &.primary {
    color: #b3e5fc;
  }
  &.secondary {
    color:  aquamarine;
  }
  &.xl {
    font-size: 20vw;
  }
  &.l {
    font-size: 20vh;
  }
  &.m {
    font-size: 10vh;
  }
  &.s {
    font-size: 2em;
  }
}
