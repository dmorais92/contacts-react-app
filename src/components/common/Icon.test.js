import React from 'react';
import ReactDOM from 'react-dom';
import Icon from './Icon';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Icon />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders with a logo', () => {
  const component = renderer.create(
    <Icon icon="logo"/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
