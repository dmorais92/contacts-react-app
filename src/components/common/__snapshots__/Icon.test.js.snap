// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with a logo 1`] = `
<i
  className="Icon fas fa-user-circle m primary"
/>
`;
